### ISSUE TITLE
ADR-101 v0.38.x : BaseService and VersionService backport

### ISSUE THREAD

**Issue by jmalicevic** [cometbft/cometbft#1421]
Backporting PR #818  to `v0.38.x-experimental` branch

Commit to backport: 65a88b9f8 #1437

Branch `v0.38.x` contains a **legacy gRPC** implementation. 

Branch `main` does not have the legacy gRPC, which has been removed, and #818 adds a new gRPC service.

When backporting #818 we have two options:

1. Remove the legacy gRPC and replace it by the new gRPC service implemented in `main`
    - For preserving the existing functionalities in `v0.38`, this approach requires implementing the existing two legacy gRPC endpoints (`BroadcastTx` and `Ping`) using the new gRPC service
2. Preserve the legacy gRPC implementation and add the new gRPC implemented in `main` as an additional service
    - In this case, 3 RPC endpoints could be available on a node: the standard JSON-RPC, the legacy gRPC, and the newly introduced gRPC service

Additional changes:
- [x] Update tests to ensure that the legacy gRPC endpoints works as before
- [x] Update e2e test infra to: in case of approach 1, add test for the legacy endpoint to ensure it works with the API; in case of approach 2, extend the e2e node to support 3 endpoints
- [x] Changelog entries explaining what is breaking compared to v0.38.x, if any. (there were none)

**Comment by cason** [comment]
We are more inclined towards option (2) because:
 - The legacy gRPC is optional, and probably not employed by most users
 - The legacy gRPC service is deprecated, being removed from `main`
 - Implementing the legacy gRPC endpoints on the new gRPC server is an additional issue by itself

**Comment by cason** [comment]
Backporting 3479642e2 at this point is not ideal, as there are other PRs associated to #1419 that should be considered first:

```
% git log --oneline rpc/grpc/ 
4f0d5b4f0 (origin/feature/adr101-pull-companion) gRPC for Tx and Block indexer pruning (#1327)
6fc2c37c8 ADR 101: Refactor height check-related logic and tests (#1271)
3479642e2 ADR 101: Add `Close` method to gRPC client (#1251)
69843bed0 (origin/jasmina/benchmark_pruning) ADR-101: implement gRPC `PruningService` (#1154)
8006a7456 ADR 101: Add GetLatest method to block service (#1209)
f9c17b45d Add BlockResults gRPC service (#1168)
84e2e01e6 Add gRPC block service (#1142)
65a88b9f8 grpc: Add base gRPC server with version service (#818)
```

**Comment by jmalicevic** [comment]
yes we should separate it in a different issue and close this one once we merge the first PR.

**Comment by jmalicevic** [comment]
Closed with #1437
